src/server/adminDeleteUser.ts

```diff
@@ -2,30 +2,78 @@
 
 import { getSupabaseAdminAuthClient } from "@/supabaseAdminAuthClient";
 import { authenticateAsAdmin } from "@/server/authenticateAdminUser";
-import { logInfoReturnLogId } from "@/logger/logger";
+import { logErrorReturnLogId, logInfoReturnLogId } from "@/logger/logger";
+import { AuditLog, sendAuditLog } from "@/server/auditLog";
+import { getSupabaseServerComponentClient } from "@/supabaseServer";
 
-export type DeleteUserErrorType = {
-    error: Record<string, string> | null;
-};
+type DeleteUserErrorType =
+    | "Failed to authenticate as admin"
+    | "Failed to get user id from profiles"
+    | "Failed to delete user";
 
-export async function adminDeleteUser(userId: string): Promise<DeleteUserErrorType> {
+type DeleteUserError =
```

Comment:
nitpick: this whole object is a result, which may or may not have an error inside
```suggestion
type DeleteUserResult =
```